src/channel/ChannelContext.cpp

```diff
@@ -265,8 +265,7 @@ void ChannelContext::EnterCasePairingState()
     Transport::PeerAddress addr;
     addr.SetTransportType(Transport::Type::kUdp).SetIPAddress(prepare.mAddress);
     CHIP_ERROR err = prepare.mCasePairingSession->EstablishSession(
-        addr, &prepare.mBuilder.GetOperationalCredentialSet(), prepare.mBuilder.GetOperationalCredentialSetIndex(),
-        prepare.mBuilder.GetPeerNodeId(), mExchangeManager->GetNextKeyId(), ctxt, this);
+        addr, mFabricsTable, mFabricIndex, prepare.mBuilder.GetPeerNodeId(), mExchangeManager->GetNextKeyId(), ctxt, this);
```

Comment:
This may have been a comment when we did opcredset/index as well: why do we pass 'fabricstable'  and 'fabricsindex'  instead of resolving to a single fabric?

For general code, if I see pass in "array + index in the array  to use", in which case I would rather expect to "pass in the element needed  in the array". Why does this not apply here? Would it be worth it to at least encapsulate a 'FabricId' that underneath contains FabricTable+index to show that these are always used in pairs?

Reply:
I think it should be ok to pass FabricInfo here. I'll try it out and update the PR if I don't see any issues with it.